### DISCUSSION TITLE
Enforce rigid body motion for a block

### DISCUSSION THREAD

**Discussion by lyyc199586** [idaholab/moose#25299] [Q&A General] [answered]
Is there a way to enforce the displacement over a whole block (subdomain) rather than on the boundary? 

I am trying to enforce rigid body motion to an object in a contact simulation

**Comment by GiudGiud** [comment]
So are you still going to solve for the displacement in that object?

If not, you could just use an auxiliary variable for the displacement and set it in the volume with for example a ParsedAux
If so, I think you could look at:
- splitting displacement into two parts, one being imposed and the other one being solved for. That might be a lot of work
- using a volumetric initial condition for the velocity (is that a variable?)

@hugary1995 or @recuero will know more about this

**Reply by lyyc199586** [reply]
I don't want to solve it, I want to enforce (preset) the displacements on the whole block

So can I use ParsedAux to compute the displacement and just assign it to disp_x?

**Reply by recuero** [reply]
I don't know that we can impose displacements directly to entire domains. You can evidently prepare the mesh such that applying Dirichlet BCs on the boundaries presets most of the displacements in the block.

Now, if you are doing contact with a boundary that has preset displacement conditions, I'd be interested to know the outcome. I don't think we have tested that case for some of our algorithms.

**Reply by maxnezdyur** [reply, marked as answer]
Using the`BoundingBoxNodeSetGenerator` you can effectively place dirichlet BCs on an entire block. For contact with that block, I think which block is primary or secondary would be important, but I don't know off the top of my head which would be better for the "rigid" block.

**Reply by lyyc199586** [reply]
should thought about it first, could just mark it all as a boundary

**Reply by maxnezdyur** [reply]
If using a penalty formulation, I think you should make the rigid body the primary block. I am not sure about the other contact methods. Let me know how that works.

**Reply by lyyc199586** [reply]
yea that's what I am doing now, at least for elastodynamics, it works fine
